### PR TITLE
Fix e-mail confirmations API not working correctly

### PR DESCRIPTION
**Pull request by Gargron**

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -801,3 +801,9 @@ DEPENDENCIES
   webpacker (~> 5.4)
   webpush (~> 0.3)
   xorcist (~> 1.1)
+
+RUBY VERSION
+   ruby 2.7.2p137
+
+BUNDLED WITH
+   2.1.4
```
Comment (ClearlyClaire) [inline Gemfile.lock:809]: Spurious changes

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]

**Comment by ClearlyClaire** [comment]
What cases does it solve?

**Comment by Gargron** [comment]
> What cases does it solve?

The API allows a user to correct e-mail typos after sign-up through an app and/or re-request a confirmation link if hasn't arrived.

**Comment by ClearlyClaire** [comment]
I know, I meant, why was that API broken that this PR solves?

File: app/controllers/api/v1/emails/confirmations_controller.rb [hunk 2]
```diff
@@ -18,4 +17,8 @@ def create
   def require_user_owned_by_application!
     render json: { error: 'This method is only available to the application the user originally signed-up with' }, status: :forbidden unless current_user && current_user.created_by_application_id == doorkeeper_token.application_id
   end
+
+  def require_user_not_confirmed!
+    render json: { error: 'Thus method is only available while the e-mail is awaiting confirmation' }, status: :forbidden if current_user.confirmed? || current_user.unconfirmed_email.blank?
```
Comment (ClearlyClaire) [inline app/controllers/api/v1/emails/confirmations_controller.rb:22]: “This method”

**Comment by Gargron** [comment]
> I know, I meant, why was that API broken that this PR solves?

Oh. `unconfirmed_email` is not filled for new accounts. Devise only fills it out when a user is confirmed and changes their e-mail. But by that point `confirmed?` is already true. So the condition was never true.

**Comment by ClearlyClaire** [comment]
Ok, in this case, shouldn't, for a confirmed account, `unconfirmed_email` be updated instead of `email`?

**Comment by Gargron** [comment]
> Ok, in this case, shouldn't, for a confirmed account, `unconfirmed_email` be updated instead of `email`?

Devise [handles it](https://github.com/heartcombo/devise/blob/c82e4cf47b02002b2fd7ca31d441cf1043fc634c/lib/devise/models/confirmable.rb#L58)

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**